timemachine/fe/rbfe.py

```diff
@@ -537,10 +537,8 @@ def estimate_relative_free_energy_via_greedy_bisection(
 
     assert len(keep_idxs) <= n_windows
 
-    if min_overlap is not None and keep_idxs not in [None, [0, -1], list(range(n_windows))]:
-        raise ValueError(
-            "when min_overlap is not None, keep_idxs must be equal to one of None, [0, -1], or list(range(n_windows))"
-        )
+    if min_overlap is not None and keep_idxs is not None and not all(i in {0, -1} for i in keep_idxs):
```

Comment:
```suggestion
    if min_overlap is not None and not all(i in {0, -1} for i in keep_idxs):
```
keep_idxs is already a list at this line, since we check len(keep_idxs) which would raise an exception if it was None.

Doesn't this remove the ability to have `keep_idxs` be anything besides `[0, -1]`? Should `[3, 5]` be invalid? Looks like the old code was similar as well, only allowing None (which we set to `[0, -1]` above https://github.com/proteneer/timemachine/pull/1115/files#diff-49ef8469d9fb5c9d39e05a0cef32710703096c53f657e55eff49d5c3ddd47b13R536), `list(range(windows))` and `[0, -1]`. Think I am missing why `keep_idxs` is dependent on the state of `min_overlap`, doesn't the min overlap get evaluated on intermediate windows that we don't necessarily store?

Reply:
> keep_idxs is already a list at this line, since we check len(keep_idxs) which would raise an exception if it was None.

Good catch, I'd missed that we already check for `None` above.

> Doesn't this remove the ability to have keep_idxs be anything besides [0, -1]? Should [3, 5] be valid?

I don't think `[3, 5]` or any indices except `0` and `-1` (first and last) are valid when adaptive bisection is enabled (i.e. `min_overlap` is not `None`), because we don't know ahead of time how many windows there will be. This was changed in #1093 as part of a fix for an issue where we'd been passing something like `keep_idxs=[0, n_windows - 1]`, which caused an `IndexError` when bisection resulted in fewer than `n_windows` windows.

This could still be relaxed a bit (e.g. we will always have at least 2 windows, so `1` might be allowed), but in general arbitrary indices aren't safe here.

Reply:
Does that then mean we are unable to retrieve frames for an intermediate window during adaptive bisection (without modifying the code)? Storing intermediate frames seems like a very useful feature for debugging if something goes awry in an intermediate lambda window.

Reply:
That's true, it would currently require changing the code. We should probably rethink this logic, as this had been intended as a quick patch for production, where we typically only keep first and last.

Could do something like ignore indices that don't exist at the end of a run, silently or with a warning? There should also probably be an option for "keep all frames", though that would be achievable with the proposed behavior by passing `range(n_windows)`.

Reply:
Adding back in the support for `range(n_windows)` would probably be good for now, that way frames can get gotten back out at the expense of disk space.

Reply:
Woops had misunderstood and thought we could just add back in the previous check for `list(range(n_windows))`. For now this is fine as it prevents an issue that already exists.

Would be good to follow up on this, as seems important to be able to get out intermediate frames if we need them.

Reply:
Tried to improve this in https://github.com/proteneer/timemachine/pull/1115/commits/3388ef2cc181e973e049cacb0e0922d57d592a88.

This reverts to the old behavior w.r.t `keep_idxs`, except that we'll now skip storing (with warning) indices that are invalid after bisection is complete. So e.g. `range(n_windows)` is valid as an input and will result in storing frames for at most `n_windows` windows.